### PR TITLE
[QF-4240] fix language switch flicker

### PR DESCRIPTION
**Pull request by afifvdin**
## Summary

Fixed theme flicker/flash issue that occurred when switching languages. The Redux store is recreated on language change, and Redux-persist's async REHYDRATE was causing a brief flash of the default theme before the persisted theme loaded. This fix synchronously reads the persisted theme from localStorage before store creation to preserve the user's theme preference immediately.

Closes: [QF-4240](https://quranfoundation.atlassian.net/browse/QF-4240)

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Environment Variables

N/A

## Test Plan

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Set the theme to dark mode (or light mode if default is dark)
2. Switch to a different language using the language switcher
3. Verify that the theme remains consistent without any flash/flicker
4. Verify that the theme preference is still correctly applied after language change
5. Test with both logged-in and logged-out users
6. Test switching between multiple languages in succession
7. Verify that the theme persists correctly on page reload after language change

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [ ] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [ ] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [x] Inline comments added for complex logic

### Localization (if UI changes)

- [ ] All user-facing text uses `next-translate`
- [ ] Only English locale files modified (Lokalise handles others)
- [ ] RTL layout verified

### Accessibility (if UI changes)

- [ ] Semantic HTML elements used
- [ ] ARIA attributes added where needed
- [ ] Keyboard navigation works

## Screenshots/Videos

| Before | After |
| ------ | ----- |
|<video src="https://github.com/user-attachments/assets/ec1a7674-f9db-4cf4-a5bd-49ab30929e83" width="300" />|<video src="https://github.com/user-attachments/assets/71d0e043-bc83-414c-8cc0-aa775b431528" width="300" />|

## Related PRs

N/A

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4240]: https://quranfoundation.atlassian.net/browse/QF-4240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/redux/utils/getPersistedTheme.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { RootState } from '../RootState';
+import SliceName from '../types/SliceName';
+
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/redux/utils/getPersistedTheme.ts:3]: yes

File: src/redux/store.ts [hunk 2]
```diff
@@ -115,8 +116,21 @@ const getStore = (
...
+    preloadedState ||
+    getStoreInitialState(locale, countryPreference, detectedLanguage, detectedCountry);
+
+  // Preserve persisted theme synchronously to prevent flash during language switching.
+  // Redux-persist's REHYDRATE is async, so we read localStorage before store creation.
+  if (typeof window !== 'undefined') {
+    const persistedTheme = getPersistedTheme();
+    if (persistedTheme) {
+      initialState[SliceName.THEME] = persistedTheme;
+    }
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/redux/store.ts]: make sense

File: src/redux/utils/getPersistedTheme.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    const parsed = JSON.parse(persistedState);
+    const themeState = parsed[SliceName.THEME];
+    if (!themeState) {
+      return null;
+    }
+
+    return JSON.parse(themeState);
+  } catch (error) {
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/redux/utils/getPersistedTheme.ts:33]: ok
